src/BuiltInTools/dotnet-watch/HotReloadDotNetWatcher.cs

```diff
@@ -186,13 +192,41 @@ public async Task WatchAsync(DotNetWatchContext context, CancellationToken cance
             }
         }
 
-        private static void ConfigureExecutable(DotNetWatchContext context, ProcessSpec processSpec)
+        private void ConfigureExecutable(DotNetWatchContext context, ProcessSpec processSpec)
         {
             var project = context.FileSet.Project;
-            processSpec.Executable = project.RunCommand;
-            if (!string.IsNullOrEmpty(project.RunArguments))
+
+            // For webapps, we can support changes to Program.Main and Startup by restarting the host (aka hot restart).
+            // First determine if we can perform hot restarts.
+            if (!SuppressHotRestart &&
+                ProjectGraph.EntryPointNodes.FirstOrDefault().ProjectInstance is { } projectInstance &&
```

Comment:
What is this validating?

Reply:
Just being extra careful. We want to make sure there's a ProjectInstance we can query against.